content/en/docs/guides/performance.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Performance Best Practices
+description: A user guide of both general and language-specific best practices to improve performance.
+---
+
+### General
+
+*   Always **re-use stubs and channels** when possible.
+
+*   **Use keepalive pings** to keep HTTP/2 connections alive during periods of
+    inactivity to allow initial RPCs to be made quickly without a delay (i.e.
+    C++ channel arg GRPC_ARG_KEEPALIVE_TIME_MS). 
+
+*   **Use streaming RPCs** when wanting to avoid continuous RPC initiation,
```

Comment:
I admit to people that steams are faster than unary RPCs, but I _discourage_ using them for performance reasons. You have be seriously hurting for performance before the costs make sense. If the unary RPCs would be _easier_ as a stream, go for it. But using a stream _just_ for the gRPC-level performance seems a mistake.

I _do_ tell people to use streams if it makes _their application_ faster, e.g., because of reduced synchronization between backends.

Reply:
Hm interesting, this bullet does mention that "Streams are particularly well-suited to handling    a long-lived logical flow of data from the client-to-server,    server-to-client, or in both directions". We could emphasize this point at the start of this bullet as the main reason to use streaming RPCs?

Reply:
Yeah, that would be fine. Mainly I'd want to downplay per-RPC overhead.

We can mention that streams have less overall overhead, but it'd be important to downplay it and mention that abusing streams might increase performance at small scale but reduce scalability due to LB issues and complexity (although obviously some rare cases are appropriate to pay those costs). Basically, I'd focus on optimizing the application architecture.

Reply:
Done, emphasized that streams should only be used for long-lived data flows and their impact on scalability